src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -55,9 +55,17 @@ private function init($redisClient, string $namespace, int $defaultLifetime, ?Ma
         if (preg_match('#[^-+_.A-Za-z0-9]#', $namespace, $match)) {
             throw new InvalidArgumentException(sprintf('RedisAdapter namespace contains "%s" but only characters in [-+_.A-Za-z0-9] are allowed.', $match[0]));
         }
+
         if (!$redisClient instanceof \Redis && !$redisClient instanceof \RedisArray && !$redisClient instanceof \RedisCluster && !$redisClient instanceof \Predis\ClientInterface && !$redisClient instanceof RedisProxy && !$redisClient instanceof RedisClusterProxy) {
             throw new InvalidArgumentException(sprintf('%s() expects parameter 1 to be Redis, RedisArray, RedisCluster or Predis\ClientInterface, %s given.', __METHOD__, \is_object($redisClient) ? \get_class($redisClient) : \gettype($redisClient)));
         }
+
+        if ($redisClient instanceof \Predis\ClientInterface && $redisClient->getOptions()->exceptions) {
+            $options = clone $redisClient->getOptions();
+            \Closure::bind(function () { $this->options['exceptions'] = false; }, $options, $options)();
+            $redisClient = new $redisClient($redisClient->getConnection(), $options);
```

Comment:
this relies on the signature of the `Predis\Client` constructor, so I don't think using a dynamic class name is a good idea here.

Reply:
See above: Predis does this internally already so we're consistent.